js/ironx.js

```diff
@@ -0,0 +1,655 @@
+        return result;
+    }
+
+    async fetchCurrencies (params = {}) {
+        const response = await this.publicGetCurrencies (params);
+        const currencies = this.safeValue (response, 'result', []);
+        const result = {};
+        for (let i = 0; i < currencies.length; i++) {
+            const currency = currencies[i];
+            const id = this.safeString (currency, 'symbol');
+            const code = this.safeCurrencyCode (id);
+            result[code] = ({
+                'info': currencies[i],
+                'id': id,
+                'code': code,
+                'name': this.safeString (currency, 'name'),
+                'active': currency['isActive'],
+                'fee': undefined,
+                'precision': this.safeInteger (currency, 'amountPrecision'),
+                'limits': {
+                    'amount': {
+                        'min': undefined,
+                        'max': undefined,
+                    },
+                    'price': {
+                        'min': undefined,
+                        'max': undefined,
+                    },
+                    'cost': {
+                        'min': undefined,
+                        'max': undefined,
+                    },
+                    'withdraw': {
+                        'min': undefined,
+                        'max': undefined,
+                    },
+                },
+            });
+        }
+        return result;
+    }
+
+    async fetchBalance (params = {}) {
+        await this.loadMarkets ();
+        const response = await this.privateGetUserBalances (params);
+        const balance = this.safeValue (response, 'result', []);
+        return this.parseBalance (balance);
+    }
+
+    parseBalance (balance) {
+        const result = { 'info': balance, 'free': {}, 'used': {}, 'total': {}};
+        const currencies = Object.keys (balance.available);
+        for (let i = 0; i < currencies.length; i++) {
+            const code = currencies[i];
+            // console.log (this.asFloat (balances.available[code]));
+            const fValue = this.asFloat (balance.available[code]);
+            const tValue = this.asFloat (balance.total[code]);
+            const uValue = this.asFloat (this.decimalToPrecision (this.sum (tValue, -fValue), 3, 6));
+            result[code] = {
+                'free': fValue,
+                'used': uValue,
+                'total': tValue,
+            };
+            result.free[code] = fValue;
+            result.used[code] = uValue;
+            result.total[code] = tValue;
+        }
+        return result;
+    }
+
+    async fetchOrderBook (symbol, limit = undefined, params = {}) {
+        await this.loadMarkets ();
+        const response = await this.publicGetBookSymbol (this.extend ({
+            'symbol': this.marketId (symbol),
+        }, params));
+        const orderbook = this.safeValue (response, 'result', []);
+        // console.log (orderbook);
```

Comment:
Please try not to push logs to production branch. If there is need for "important" debugging wrap in **try/catch**.

Reply:
Sure sorry for that ... cleaning up right now.